app/tests/unit/services/bucketPermission.spec.js

```diff
@@ -0,0 +1,80 @@
+const { MockModel, MockTransaction } = require('../../common/dbHelper');
+const { NIL: SYSTEM_USER } = require('uuid');
+
+jest.mock('../../../src/db/models/tables/objectPermission', () => MockModel);
+jest.mock('../../../src/db/models/tables/bucketPermission', () => MockModel);
+
+const { Permissions } = require('../../../src/components/constants');
+
+const service = require('../../../src/services/bucketPermission');
+
+const bucketId = '00000000-0000-0000-0000-000000000000';
+const objectId = '00000000-0000-0000-0000-000000000000';
+const userId = '00000000-0000-0000-0000-000000000000';
+
+const data = [{
+  id: objectId,
+  bucketId: bucketId,
+  path: 'path',
+  public: 'true',
+  active: 'true',
+  createdBy: userId,
+  permCode: 'READ'
+}];
+
+const params = {
+  bucketId: bucketId,
+  permCode: 'READ',
+  userId: userId
+};
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('searchPermissions', () => {
+
+  it('should filter and return bucket permissions', async () => {
+    service.searchPermissions(params);
+    expect(MockModel.query).toHaveBeenCalledTimes(1);
+  });
+});
+
+describe('addPermissions', () => {
+
+  it('should search and return bucket permissions', async () => {
+    // add userId property
+    const dataWithUser = { ...data, userId: userId };
+
+    const perms = Object.values(Permissions).map((p) => ({
```

Comment:
i dont think we need this perms object. searchPermissionSpy should return bucket permissions in a different format.
eventually we will want to test the code between line 34 and 48 of the addPermission() function.
maybe just mock everything in this addPermissions function and dont bother testing anything except that mockeModel was called.